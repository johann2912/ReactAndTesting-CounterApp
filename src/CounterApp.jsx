import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    /**
     * add values for counter
     */
    const handleAdd = () => {
        setCounter(counter + 1);
    }

    /**
     * delete values for counter
     */
    const handleSubstract = () => setCounter(counter - 1);


    /**
     * reset count for counter 
     */
    const handleReset = () => setCounter(value);


    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button aria-label='btn-add' onClick={handleAdd} >
                +1
            </button>
            <button aria-label='btn-reset' onClick={handleReset} >
                Reset
            </button>
            <button aria-label='btn-substrac' onClick={handleSubstract} >
                -1
            </button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
    value: 'Does not found value'
}