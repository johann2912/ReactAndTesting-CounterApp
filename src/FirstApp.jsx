import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle, name } ) => {

    

    return (
        <>
            <h1 data-testid="test-title" > { title } </h1>
            <p> { subTitle } </p>
            <p> { subTitle } </p>
            <p> { name } </p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    name: 'Kakaroto',
    subTitle: 'Sub-title does not found',
    //title: 'Title does not found',
}