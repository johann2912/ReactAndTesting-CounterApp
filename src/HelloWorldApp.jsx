export const HelloWorldApp = () => {
    /**
     * document.createElement...
     */
    return (
        <h1>Hello world app</h1>
    );
};

//export default App;