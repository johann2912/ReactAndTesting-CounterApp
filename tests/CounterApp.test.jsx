import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from '../src/CounterApp';

describe('Test <CounterApp />', () => {

    const initialValue = 23;

    test('Should match with snapshot', () => {
        const { container } = render(<CounterApp value={ initialValue } />)

        expect(container).toMatchSnapshot();
    })

    test('Should show value initial is an 100 <CounterApp value={100} />', () => {
        render(<CounterApp value={ 100 } />)

        expect(screen.getByText(100)).toBeTruthy();
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100')
    })

    test('increment button +1', () => {
        render(<CounterApp value={ initialValue } />);
        fireEvent.click(screen.getByRole('button', { name: 'btn-add'}));
        expect(screen.getByText('24')).toBeTruthy();
    })

    test('decrement button -1', () => {
        render(<CounterApp value={ initialValue } />);
        fireEvent.click(screen.getByRole('button', { name: 'btn-substrac'}));
        expect(screen.getByText('22')).toBeTruthy();
    })

    test('reset button', () => {
        render(<CounterApp value={ initialValue } />);
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset'}));
        
        expect(screen.getByText('23')).toBeTruthy();
    })

})