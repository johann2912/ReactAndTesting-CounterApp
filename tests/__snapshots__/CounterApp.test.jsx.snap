// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CounterApp /> Should match with snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    23
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button
    aria-label="btn-substrac"
  >
    -1
  </button>
</div>
`;
